mobcat_shared/MobCAT/MVVM/BaseShellViewModel.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.MobCAT.MVVM.Abstractions;
+
+namespace Microsoft.MobCAT.MVVM
+{
+    public class BaseShellViewModel : BaseNavigationViewModel
+    {
+        private IShellNavigationService _shellNavigationService;
+        protected IShellNavigationService ShellNavigationService
+        {
+            get
+            {
+                if (_shellNavigationService == null)
+                {
+                    _shellNavigationService = Navigation as IShellNavigationService;
```

Comment:
Suggestion: Should we just force the Navigation to be an instance of IShellNavigationService? BaseShellViewModel should assume that navigation is IShellNavigationService and accessible otherwise just prevent a user from accessing ShellNavigationService property and just throw an exception.

not a blocked to merge

Reply:
So I'm clear, in that case: BaseShellViewModel would resolve IShellNavigationService and the user would need to register both INavigationService and IShellNavigationService since BaseNavigationViewModel uses INavigationService instead of registering IShellNavigationService as INavigationService right? It feels kind of redundant because ShellNavigationService already inherits from NavigationService. 

Currently
`            ServiceContainer.Register<INavigationService>(shellNavigationService);`

That being said, it might be clearer to consumers of MobCAT. What's everyones else's thoughts?

Reply:
@alexkblount fixed this by overriding the Navigation property in BaseShellViewModel to resolve the IRouteNavigationService. Great solve! Please take a look at the new implementation in BaseShellViewModel and BaseNavigationViewModel which was updated to support BaseContentPage since there's no functionality to justify a new BaseShellContentPage